spec/h2_connection_spec.lua

```diff
@@ -157,6 +159,42 @@ describe("http2 connection", function()
 			assert.truthy(cq:empty())
 		end)
 	end)
+	describe("priority on missing stream", function()
+		it("sets default priority for streams with missing parent", function()
+			local cq = cqueues.new()
+			local c, s = cs.pair()
+			cq:wrap(function()
+				c = h2_connection.new(c, "client")
+				local client_stream = c:new_stream()
+				local req_headers = new_headers()
+				req_headers:append(":method", "GET")
+				req_headers:append(":scheme", "http")
+				req_headers:append(":path", "/")
+				assert(client_stream:write_headers(req_headers, false))
+				assert(c:close())
+			end)
+			cq:wrap(function()
+				s = h2_connection.new(s, "server")
+				-- Mock HEADER frame handler to inject false stream priority
+				local saved_handler = h2_stream.frame_handlers[0x1]
+				h2_stream.frame_handlers[0x1] = function (stream, flags, payload)
+					flags, payload = flags + 0x20, spack("> I4 B", 99, 99)..payload
+					return saved_handler(stream, flags, payload)
+				end
+				-- Check if not set to mocked priority
+				local ok, err = pcall(function()
```

Comment:
yielding over `pcall` doesn't work in lua 5.1 so you unfortunately can't use it here.